docs_nnx/guides/flax_gspmd.md

```diff
@@ -53,14 +58,14 @@ import optax # Optax for common losses and optimizers.
 ```
 
 ```{code-cell} ipython3
-print(f'We have 8 fake JAX devices now: {jax.devices()}')
+print(f'You have 8 “fake” JAX devices now: {jax.devices()}')
 ```
 
 The code below shows how to import and set up the JAX-level device API, following JAX's [Distributed arrays and automatic parallelization](https://jax.readthedocs.io/en/latest/notebooks/Distributed_arrays_and_automatic_parallelization.html) guide:
 
-1. Start a 2x4 device `mesh` (8 devices) using JAX's `mesh_utils.create_device_mesh`. This layout is the same as the one of a [TPU v3-8](https://cloud.google.com/tpu/docs/system-architecture-tpu-vm#single_tpu_board).
+1. Start a 2x4 device `mesh` (8 devices) using the JAX [`mesh_utils.create_device_mesh`](https://jax.readthedocs.io/en/latest/_autosummary/jax.experimental.mesh_utils.create_device_mesh.html). This layout is the same as the one of a [TPU v3-8](https://cloud.google.com/tpu/docs/system-architecture-tpu-vm#single_tpu_board).
 
-2. Annotate each axis with a name using the `axis_names` parameter in `jax.sharding.Mesh`. A typical way to annotate axis names is `axis_name=('data', 'model')`, where:
+2. Annotate each axis with a name using the `axis_names` parameter in [`jax.sharding.Mesh`](https://jax.readthedocs.io/en/latest/jax.sharding.html#jax.sharding.Mesh). A typical way to annotate axis names is `axis_name=('data', 'model')`, where:
```

Comment:
Since we mentioned `jax.sharding.Mesh` above, just write this:

```
2. Annotate each axis with a name using the `axis_names` parameter. A typical way to annotate axis names is `axis_name=('data', 'model')`, where:
```

Reply:
SGTM!